File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.cpp
class Solution {
public:
    double minimumAverage(vector<int>& nums) {
        int i = 0, j = nums.size()-1;
        sort(nums.begin(), nums.end());
        double ans = INT_MAX;
        while(i < j) {
            double val = (double)(nums[i] + nums[j])/2;
            ans = min(ans, val);
            i++; j--;
        }
        return ans;
    }
};

File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.java
class Solution {
    public double minimumAverage(int[] nums) {
        Arrays.sort(nums);
        int i = 0, j = nums.length-1;
        double val = 0, ans = (double)Integer.MAX_VALUE;
        while(i < j) {
            val = (double)(nums[i] + nums[j])/2;
            ans = Math.min(val, ans);
            i++;
            j--;
        }
        return ans;
    }
}